trace_decoder/src/decoding.rs

```diff
@@ -618,6 +628,47 @@ fn process_txn_info(
     Ok(gen_inputs)
 }
 
+fn merge_batch_jumpdest_tables<'t, T>(jdts: T) -> Option<JumpDestTableWitness>
+where
+    T: Iterator<Item = &'t Option<JumpDestTableWitness>>,
+{
+    let mut merged_table = JumpDestTableWitness::default();
+
+    let mut max_batch_ctx = 0;
+    for jdt in jdts {
+        let tx_offset = max_batch_ctx;
+        // abort if any transaction in the batch came without RPC JumpDestTable.
+        if jdt.is_none() {
+            return None;
+        }
+        for (code_hash, ctx_tbl) in jdt.as_ref().unwrap().0.iter() {
+            for (ctx, jumpsdests) in ctx_tbl.0.iter() {
+                let batch_ctx = tx_offset + ctx;
+                max_batch_ctx = max(max_batch_ctx, batch_ctx);
+
+                merged_table
+                    .0
+                    .entry(*code_hash)
+                    .or_insert(ContextJumpDests::default());
+
+                merged_table
+                    .0
+                    .get_mut(code_hash)
+                    .unwrap()
+                    .0
+                    .entry(batch_ctx)
+                    .or_insert(jumpsdests.clone());
```

Comment:
```suggestion
                merged_table
                    .0
                    .entry(*code_hash)
                    .and_modify(|ctx_jdts| {
                        ctx_jdts.0.insert(batch_ctx, jumpsdests.clone());
                    })
                    .or_insert(ContextJumpDests(HashMap::from([(
                        batch_ctx,
                        jumpsdests.clone(),
                    )])));
```

Reply:
(side note, it'd be simpler to define `ContextJumpDests` as a type alias instead of a wrapper around a pub `HashMap`)

Reply:
I have a medium preference for [New Type Idiom](https://doc.rust-lang.org/rust-by-example/generics/new_types.html) over type aliases, as the latter don't help you catch type errors.